### PR TITLE
feature: HTTP3 authority as http_host header.

### PR DESCRIPTION
**Pull request by zhuizhuhaomeng**
I hereby granted the copyright of the changes in this pull request
to the authors of this lua-nginx-module project.

### CODE REVIEW

File: src/ngx_http_lua_headers.c [hunk 1]
```diff
@@ -793,12 +798,53 @@ ngx_http_lua_ffi_req_get_headers_count(ngx_http_request_t *r, int max,
...
+                if (part->next == NULL) {
+                    break;
+                }
+
+                part = part->next;
+                header = part->elts;
+                i = 0;
+            }
+
+            if (has_host == 1 && header[i].key.len == 4
```
Comment (lynch1981) [inline src/ngx_http_lua_headers.c:825]: duplicate check has_host

File: src/ngx_http_lua_headers.c [hunk 2]
```diff
@@ -793,12 +798,53 @@ ngx_http_lua_ffi_req_get_headers_count(ngx_http_request_t *r, int max,
         max = NGX_HTTP_LUA_MAX_HEADERS;
     }
 
+    count = 0;
     part = &r->headers_in.headers.part;
+    header = part->elts;
```
Comment (lynch1981) [inline src/ngx_http_lua_headers.c:803]: header should be wrapped in NGX_HTTP_V3

File: src/ngx_http_lua_subrequest.c [hunk 3]
```diff
@@ -1699,6 +1701,16 @@ ngx_http_lua_copy_request_headers(ngx_http_request_t *sr,
     part = &pr->headers_in.headers.part;
     header = part->elts;
 
+#if (NGX_HTTP_V3)
+    if (pr->headers_in.server.data != NULL) {
+        if (ngx_http_lua_set_input_header(sr, host_header,
+                                         pr->headers_in.server, 0) == NGX_ERROR)
```
Comment (swananan) [inline src/ngx_http_lua_subrequest.c:1707]: code style, parameters are not aligned

Comment (zhuizhuhaomeng) [reply src/ngx_http_lua_subrequest.c:1707]: This line hit the 80-character limit.

Comment (swananan) [reply src/ngx_http_lua_subrequest.c:1707]: It appears that the nginx core addresses this situation in the following way.
```suggestion
        if (ngx_http_lua_set_input_header(sr, host_header,
                                                                     pr->headers_in.server, 0)
             == NGX_ERROR)
```

**Pull request merged**